README.md

```diff
@@ -0,0 +1,262 @@
+# Nushell RFCs
+
+[Nushell RFCs]: #nushell-rfcs
+
+Many changes, including bug fixes and documentation improvements can be
+implemented and reviewed via the normal GitHub pull request workflow.
+
+Though ssome changes are substantial enough that we ask for these to be
+put through a bit of a design process and produce a consensus among the
+Nushell community.
+
+The "RFC" (request for comments) process is intended to provide a consistent
+and controlled path for new features to be added to Nushell, so that all
+stakeholders can be confident about the direction the project is evolving in.
+
+
+## Table of Contents
+[Table of Contents]: #table-of-contents
+
+  - [Opening](#nushell-rfcs)
+  - [Table of Contents]
+  - [When you need to follow this process]
+  - [Before creating an RFC]
+  - [What the process is]
+  - [The RFC life-cycle]
+  - [Reviewing RFCs]
+  - [Implementing an RFC]
+  - [RFC Postponement]
+  - [Help this is all too informal!]
+  - [License]
+  - [Contributions]
+  - [Acknowledgements]
+
+
+## When you should follow this process
+[When you should follow this process]: #when-you-should-follow-this-process
+
+*Note: While we establish this RFC process, this says "should"; eventually
+it should say "need to".*
+
+You should follow this process if you intend to make substantial changes to
+Nushell, or this RFC process itself. What constitutes a substantial change is
+evolving based on community norms and varies depending on what part of the
+ecosystem you are proposing to change, but may include the following:
+
+  - Any semantic or syntactic change that is not a bugfix.
+  - Removing existing features.
+  - Changes to internal interfaces, like the plugin architecture.
+
+Some changes do not require an RFC:
+
+  - Rephrasing, reorganizing, refactoring, or otherwise "changing shape does
+    not change meaning".
+  - Additions that strictly improve objective, numerical quality criteria
+    (warning removal, speedup, better platform coverage, more parallelism,
+    better handle errors, etc.)
+  - Additions only likely to be _noticed by_ other developers of Nushell,
+    invisible to users of Nushell.
+
+If you submit a pull request to implement a new feature without going through
+the RFC process, it may be closed with a polite request to submit an RFC first.
+
+
+## Before creating an RFC
+[Before creating an RFC]: #before-creating-an-rfc
+
```

Comment:
I feel like there's something the Nu team(s) should do to help the RFC process. This paragraph assumes that the person writing it understands what "hastily-proposed RFC" means.

One possible thing we should talk about is to help direct people towards mentors who can help refine the ideas so that they cover the necessary steps in a way that doesn't read as hasty, but instead fits better with the overall direction and style of RFCs.

Something else that I like are the separate stages. In the Rust RFC process, I felt like we would sometimes jump into talking about the details of a solution before we'd really agreed on the exact problem being solved. Making the stages clear, and having a way for people to understand how to move between the stages could also be helpful for both reviewers and writers.

Reply:
> One possible thing we should talk about is to help direct people towards mentors [...]

Given the current state, effectively that would mean "chat with us on Discord in #design-discussion", wouldn't it?

If not, could you go into more detail?

> [stages]

Do you have something like the TC39 stages in mind? https://tc39.es/process-document/ At least I think those could be a good reference.

I don't think we should start with that, though. I'd rather leave it open to adopt stages like that later.

We can replace some of the language here, like "hastily" and make "strawperson" (from TC39) RFCs more welcome.

Reply:
Good points. Yes, agreed that "please come talk to us on discord" is probably the right first step :)

And right, I was thinking something like tc39 stages (I think ember uses something similar, too, though I'm less familiar with their process). Agreed that's too much for this stage. Right now just agreeing on the problem and having the RFC being a rough sketch of what they want to try should be plenty for us to use.

Reply:
Looking at https://github.com/emberjs/rfcs I don't see well-defined stages. It looks very similar to Rust's process.

> Good points. Yes, agreed that "please come talk to us on discord" is probably the right first step :)

Its kinda in this section already, but a bit buried, and without mentioning the channel. I'll rework that.

Reply:
I've dropped the entire first paragraph (no more mention of "hastily") and edited the others a bit.